lib/src/androidTest/java/com/couchbase/lite/internal/core/C4QueryBaseTest.java

```diff
@@ -100,12 +100,14 @@ protected C4Query compile(String whereExpr, String sortExpr, boolean addOffsetLi
     protected List<String> run(Map<String, Object> params) throws LiteCoreException {
         List<String> docIDs = new ArrayList<>();
         C4QueryOptions opts = new C4QueryOptions();
-        C4QueryEnumerator e = query.run(opts, encodeParameters(params));
+        AllocSlice encodedParams = encodeParameters(params);
+        C4QueryEnumerator e = query.run(opts, encodedParams);
         assertNotNull(e);
         while (e.next()) {
             docIDs.add(e.getColumns().getValueAt(0).asString());
         }
-        e.free();
+        if (e != null) e.free();
```

Comment:
assertNotNill(e) already doing, is _e != null_ necessary?

Reply:
I will rewrite this a bit.